### ISSUE TITLE
Do not require explicitly setting the baseline version in two places

### ISSUE THREAD

**Issue by leeavital** [palantir/gradle-baseline#216]
Users need to specify the baseline version in two places:

```
buildscript {

   classpatch "baseline:<version>
}

// and then again
dependencies {
  baseline "baseline:<version>@zip"
}
```

users are choosing to either factor out a variable into gradle.properties or leave a comment reminding maintainers to keep the values in sync. Both of these don't gel with our gradle linter which expects version numbers to live in a version.props file. If would be great if baseline could automatically pull down the right version of the baseline configuration if the explicit setting in the dependencies block was ommited.

**Comment by markelliot** [comment]
There’s not really a great way to version the buildscript dependencies since the recommender is itself a buildscript dependency.

**Comment by leeavital** [comment]
I think it's fine to version the buildscript dependency explicitly.  But I think it should be possible to automatically configure the `baseline` dependency that accompanies it. Something like:

```
afterEvaluate {
    if (!baselineConfigured()) {
       project.configurations.add("baseline").add("baseline:" + getBaselineVersion() + "@zip");
   }
}
```

in the plugin (I may be mixed up about gradle lifecycle things)

**Comment by jmcampanini** [comment]
configurations have a concept of `defaultDependencies` which lets plugin authors add specific dependencies if none are provided: https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/Configuration.html#defaultDependencies(org.gradle.api.Action)

**Comment by qinfchen** [comment]
A related ticket, https://github.com/palantir/gradle-baseline/issues/193.  

> The default was removed in order to make the configuration space more
democratic and not favour a single configuration.

**Comment by iamdanfox** [comment]
Fixed in [0.30.0](https://github.com/palantir/gradle-baseline/releases/tag/0.30.0)
